### PR TITLE
Enable the new model serialization for Advisor

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: eng/templates/Azure.ResourceManager.Template/src/autorest.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ sample-gen:
 skip-csproj: true
 modelerfour:
   flatten-payloads: false
+use-model-reader-writer: true
```
Comment (live1206) [inline eng/templates/Azure.ResourceManager.Template/src/autorest.md:19]: @ArcturusZhang When are we planning to remove this flag to enable public serialization by default?

**Review by live1206 — APPROVED** [review]

File: eng/templates/Azure.ResourceManager.Template/src/autorest.md (hunk 1 above)
Comment (ArthurMa1978) [reply eng/templates/Azure.ResourceManager.Template/src/autorest.md:19]: I think default false is good.

Comment (live1206) [reply eng/templates/Azure.ResourceManager.Template/src/autorest.md:19]: I think the final goal is to enable this be default, and we can simplify the code quite a bit.

Comment (ArcturusZhang) [reply eng/templates/Azure.ResourceManager.Template/src/autorest.md:19]: Yeah the plan is when we have majority of the libraries enabled on this feature, we will remove this flag, and making it default on.

Comment (live1206) [reply eng/templates/Azure.ResourceManager.Template/src/autorest.md:19]: Why do we need to wait for majority of the SDKs to be enabled? 
We are working on the enabling of this for all MPG RPs.
If we remove it now, we don't need to add the flag to autorest.md.

**Pull request merged**